### PR TITLE
Show disarm reason on OSD

### LINKED ISSUE #2179: Show disarm reason in OSD
For boards with built-in OSD it would be good to see a reason of disarming on statistics screen. This will make "disarm in air" so much easier to troubleshoot.

### PR DESCRIPTION
**Pull request by digitalentity**
Fixes #2179

### CODE REVIEW

File: src/main/io/osd.c [hunk 1]
```diff
@@ -865,6 +867,9 @@ static void osdShowStats(void)
     flyMinutes %= 60;
     tfp_sprintf(buff, "%02u:%02u:%02u", flyHours, flyMinutes, flySeconds);
     displayWrite(osdDisplayPort, statValuesX, top++, buff);
+
+    displayWrite(osdDisplayPort, 2, top, "DISARM REASON    :");
```
Comment (fiam) [inline src/main/io/osd.c]: What about "DISARMED BY" rather than "DISARM REASON"?

Comment (digitalentity) [reply src/main/io/osd.c]: Good idea. Looks better

File: src/main/io/osd.c [hunk 2]
```diff
@@ -811,60 +812,64 @@ static void osdUpdateStats(void)
...
 {
+    const char * disarmReasonStr[DISARM_REASON_COUNT] = { "UNKNOWN", "TIMEOUT", "STICKS", "SWITCH", "SWITCH", "KILLSW", "FAILSAFE", "NAV SYS" };
     uint8_t top = 2;
-    const uint8_t statValuesX = 21;
+    const uint8_t statValuesX = 20;
     char buff[10];
 
     displayClearScreen(osdDisplayPort);
-    displayWrite(osdDisplayPort, 2, top++, "  --- STATS ---");
+    displayWrite(osdDisplayPort, 1, top++, "  --- STATS ---");
```
Comment (fiam) [inline src/main/io/osd.c]: Now that you're changing this, maybe we can take the opportunity to use a constant? e.g. `statNamesX`

**Review by fiam — COMMENTED** [review]
This looks pretty good. So many improvements to the OSD coming in 1.8!

File: src/main/io/osd.c (hunk 2 above)
Comment (digitalentity) [reply src/main/io/osd.c]: Done

Comment (fiam) [reply src/main/io/osd.c]: Thanks!

**Review by fiam — APPROVED** [review]

**Comment by teckel12** [comment]
Love this, great job!

**Pull request merged**